gateway/test/collection_mgmt_test.go

```diff
@@ -0,0 +1,268 @@
+		BucketName: bucketName,
+		ScopeName:  scopeName,
+	}, grpc.PerRPCCredentials(s.basicRpcCreds))
+	requireRpcSuccess(s.T(), scopeResp, err)
+
+	s.Run("CreateInheritBucketExpiry", func() {
+		colName := uuid.NewString()[:6]
+		colResp, err := colClient.CreateCollection(context.Background(), &admin_collection_v1.CreateCollectionRequest{
+			BucketName:     bucketName,
+			ScopeName:      scopeName,
+			CollectionName: colName,
+		}, grpc.PerRPCCredentials(s.basicRpcCreds))
+		requireRpcSuccess(s.T(), colResp, err)
+
+		listResp, err := colClient.ListCollections(context.Background(), &admin_collection_v1.ListCollectionsRequest{
+			BucketName: bucketName,
+		}, grpc.PerRPCCredentials(s.basicRpcCreds))
+		requireRpcSuccess(s.T(), listResp, err)
+
+		var found *admin_collection_v1.ListCollectionsResponse_Collection
+		for _, s := range listResp.Scopes {
+			if s.Name == scopeName {
+				for _, c := range s.Collections {
+					if c.Name == colName {
+						found = c
+					}
+				}
+			}
+		}
+
+		if assert.NotNil(s.T(), found, "Did not find collection on cluster") {
+			assert.Nil(s.T(), found.MaxExpirySecs)
+		}
+
+		deleteResp, err := colClient.DeleteCollection(context.Background(), &admin_collection_v1.DeleteCollectionRequest{
+			BucketName:     bucketName,
+			ScopeName:      scopeName,
+			CollectionName: colName,
+		}, grpc.PerRPCCredentials(s.basicRpcCreds))
+		requireRpcSuccess(s.T(), deleteResp, err)
+	})
+
+	s.Run("CreateWithMaxExpiry", func() {
+		colName := uuid.NewString()[:6]
+		maxExpiry := uint32(100)
+		colResp, err := colClient.CreateCollection(context.Background(), &admin_collection_v1.CreateCollectionRequest{
+			BucketName:     bucketName,
+			ScopeName:      scopeName,
+			CollectionName: colName,
+			MaxExpirySecs:  &maxExpiry,
+		}, grpc.PerRPCCredentials(s.basicRpcCreds))
+		requireRpcSuccess(s.T(), colResp, err)
+
+		listResp, err := colClient.ListCollections(context.Background(), &admin_collection_v1.ListCollectionsRequest{
+			BucketName: bucketName,
+		}, grpc.PerRPCCredentials(s.basicRpcCreds))
+		requireRpcSuccess(s.T(), listResp, err)
+
+		var found *admin_collection_v1.ListCollectionsResponse_Collection
+		for _, s := range listResp.Scopes {
+			if s.Name == scopeName {
+				for _, c := range s.Collections {
+					if c.Name == colName {
+						found = c
+					}
+				}
+			}
+		}
+
+		if assert.NotNil(s.T(), found, "Did not find collection on cluster") {
+			assert.Equal(s.T(), *found.MaxExpirySecs, maxExpiry)
+		}
+
+		deleteResp, err := colClient.DeleteCollection(context.Background(), &admin_collection_v1.DeleteCollectionRequest{
+			BucketName:     bucketName,
+			ScopeName:      scopeName,
+			CollectionName: colName,
+		}, grpc.PerRPCCredentials(s.basicRpcCreds))
+		requireRpcSuccess(s.T(), deleteResp, err)
+	})
+
+	s.Run("CreateNoExpiryUnsupportedByAPIVersion", func() {
+		colName := uuid.NewString()[:6]
+		maxExpiry := uint32(0)
+
+		apiVer := map[string]string{
```

Comment:
Would probably be good to have a helper for doing this kind of thing (and in fact, there might already be one that we use in the XApiVersion tests), but that's not needed for this PR.